File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
     int start = 0;
     int end = nums.length-1;
     int index = -1;


     while(start<=end){
        int mid = start +  (end-start)/2;
        if(nums[mid] == target){
            return mid;
        }else if(target > nums[mid]){
                // in left sub array
                if(nums[mid] >= nums[start]){
                start = mid + 1;
                }else{
                // in right sub array
                if(target<=nums[end]){
                    start = mid+1;
                }else{
                    end = mid-1;
                    }
                }
        }else{
                // in left sub array
                if(nums[mid] >= nums[start]){
                    if(target<nums[start]){
                        start = mid+1;
                    }else{
                        end = mid-1;
                    }
                }else{
                // in right sub array
                end = mid-1;
                }
        }
     }


     return index;

}
}

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:


        sub_sets: List[List[int]] = []

        def calculate_sub(index,current_list : List[int],nums: List[int]):
            if index == len(nums):
                sub_sets.append(list(current_list))
            else:
                current_list.append(nums[index])
                calculate_sub(index+1,current_list,nums)
                current_list.pop()
                calculate_sub(index+1,current_list,nums)
        

        calculate_sub(0,[],nums)

        return sub_sets



        